subprojects/logging/src/main/java/org/gradle/internal/logging/console/WorkInProgressRenderer.java

```diff
@@ -103,87 +99,59 @@ private void resizeTo(int newBuildProgressLabelCount) {
         for (int i = newBuildProgressLabelCount - 1; i >= previousBuildProgressLabelCount; --i) {
             unusedProgressLabels.push(progressArea.getBuildProgressLabels().get(i));
         }
-
-        // Try to empty the unassigned progress operations
-        while (!unusedProgressLabels.isEmpty() && !unassignedProgressOperations.isEmpty()) {
-            attach(unassignedProgressOperations.pop());
-        }
     }
 
     private void attach(ProgressOperation operation) {
```

Comment:
I restructured the code in this file to make it easier to maintain but the diff is just horrible. Maybe using split diff will be easier for the review.